OpenRA.Mods.Common/Traits/World/ActorMap.cs

```diff
@@ -359,9 +359,10 @@ bool AnyActorsAt(MPos uv, CPos a, CellLayer<InfluenceNode> layer, SubCell sub, b
 					if (checkTransient)
 						return true;
 
-					var pos = i.Actor.TraitOrDefault<IPositionable>();
-					if (pos == null || !pos.IsLeavingCell(a, i.SubCell))
-						return true;
+					// PERF: Avoid trait lookup
+					if (i.Actor.OccupiesSpace is IPositionable pos)
+						if (pos == null || !pos.IsLeavingCell(a, i.SubCell))
```

Comment:
pos can't be null here, also please remove the second if statement

Reply:
also the logic is all wrong now

Reply:
You are right. It is actually inverted.